src/scripting_api/aform.js

```diff
@@ -140,11 +140,26 @@ class AForm {
     return date;
   }
 
-  _parseDate(cFormat, cDate) {
+  _parseDate(cFormat, cDate, strict = false) {
     let date = null;
     try {
       date = this._util.scand(cFormat, cDate);
     } catch {}
+    if (!date) {
+      if (strict) {
+        return null;
+      }
+      let format = cFormat;
+      if (format.search(/mm(?!m)/) !== -1) {
```

Comment:
Nit: Here, and also below, since we don't care about the position but only want to know if something exists how about the following?
```suggestion
      if (/mm(?!m)/.test(format)) {
```